src/SpinWaveTheory/SpinWaveTheory.jl

```diff
@@ -8,9 +8,11 @@ struct SWTDataDipole
 end
 
 struct SWTDataSUN
-    s̃_mat :: Array{ComplexF64, 4}  # Dipole operators
-    T̃_mat :: Array{ComplexF64, 3}  # Single-ion anisos
-    Q̃_mat :: Array{ComplexF64, 4}  # Quadrupolar operators
+    dipole_operators :: Array{ComplexF64, 4}
+    onsite_operator :: Array{ComplexF64, 3}  # Single-ion anisotropy
+    quadrupole_operators :: Array{ComplexF64, 4}
+    sun_basis :: Array{ComplexF64, 4}
```

Comment:
Yes, it can be removed.

Reply:
Re: the other refactor, the custom observable code is actually much less complicated compared to the previous code, which was effectively using "magic numbers" (1,2,3) to specify the spin observables instead of accepting arbitrary observable matrices and using those. I recommend the new code (here and in the `lswt_speedup`) as a base for any future refactors

Reply:
Yeah, completely agree what you have is a big improvement. Regarding complications, I was only referring to leaving out the SU(N) basis for now.